peerdid/peer_did.py

```diff
@@ -24,23 +28,64 @@ def create_peer_did_numalgo_0(inception_key: PublicKeyAuthentication) -> PEER_DI
     (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
     For this type of algorithm did_doc can be obtained from peer_did
     :param inception_key: the key that creates the DID and authenticates when exchanging it with the first peer
+    :raises TypeError: if the inception_key is not instance of PublicKeyAuthentication
+    :raises ValueError: if the inception_key is not correctly encoded
     :return: generated peer_did
     """
-    pass
+    if not isinstance(inception_key, PublicKeyAuthentication):
+        raise TypeError(f"Wrong type of inception_key: {str(type(inception_key))}. Expected: PublicKeyAuthentication")
+    if not _check_key_correctly_encoded(inception_key.encoded_value, inception_key.encoding_type):
+        raise ValueError(f"Inception key is not correctly encoded")
+    peer_did = 'did:peer:0z' + _create_encnumbasis(inception_key)
+    return peer_did
 
 
 def create_peer_did_numalgo_2(encryption_keys: List[PublicKeyAgreement], signing_keys: List[PublicKeyAuthentication],
-                              service_endpoint: str) -> PEER_DID:
+                              services: JSON) -> PEER_DID:
     """
     Generates peer_did according to the second algorithm
     (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
     For this type of algorithm did_doc can be obtained from peer_did
     :param encryption_keys: list of encryption keys
     :param signing_keys: list of signing keys
-    :param service_endpoint: JSON string conforming to the DID specification (https://www.w3.org/TR/did-core/#services)
+    :param services: JSON string conforming to the DID specification (https://www.w3.org/TR/did-core/#services)
```

Comment:
I would name the parameter just `service` (w/o s) to match the name of the field in DID spec.